### PR TITLE
fix(ui-shell): add header router changes from v3

### PR DESCRIPTION
**Pull request by cal-smith**
Closes IBM/carbon-components-angular#779

### CODE REVIEW

File: src/ui-shell/header.component.ts [hunk 1]
```diff
@@ -9,13 +26,17 @@ import { I18n } from "./../i18n/i18n.module";
...
 				[href]="skipTo"
 				tabindex="0">
 				{{ i18n.get("UI_SHELL.SKIP_TO") | async }}
 			</a>
 			<ng-content select="ibm-hamburger"></ng-content>
-			<a class="bx--header__name" href="#">
+			<a
+				class="bx--header__name"
+				href="#"
+				(click)="navigate($event)">
```
Comment (rapzo) [inline src/ui-shell/header.component.ts]: Sorry for the intrusion here but wouldn't it be better to have `(selected)="navigate($event)"` to have the full hamburger functionality available for test?
I'm still getting my kicks around angular so i don't even know if that's the right MO.

Comment (cal-smith) [reply src/ui-shell/header.component.ts]: `click` is the standard HTML click event :+1: `selected` is just our convention for generic item selection events

Comment (cal-smith) [reply src/ui-shell/header.component.ts]: (And in many places we're shifting towards `value` + `valueChange` or `change` where that makes sense)

Comment (rapzo) [reply src/ui-shell/header.component.ts]: Got it. Thanks.

**Pull request merged**